pkg/ingester/ingester.go

```diff
@@ -1035,31 +1065,46 @@ func (i *Ingester) startPushRequest(ctx context.Context, reqSize int64) (_ conte
 	return ctx, true, nil
 }
 
-func (i *Ingester) finishPushRequest(reqSize int64) {
+func (i *Ingester) finishPushRequest(ctx context.Context, reqSize int64, duration time.Duration, err error) {
 	i.inflightPushRequests.Dec()
 	if reqSize > 0 {
 		i.inflightPushRequestsBytes.Sub(reqSize)
 	}
+	_ = i.circuitBreaker.finishPushRequest(ctx, duration, err)
 }
 
 // PushWithCleanup is the Push() implementation for blocks storage and takes a WriteRequest and adds it to the TSDB head.
-func (i *Ingester) PushWithCleanup(ctx context.Context, req *mimirpb.WriteRequest, cleanUp func()) error {
+func (i *Ingester) PushWithCleanup(ctx context.Context, req *mimirpb.WriteRequest, cleanUp func()) (err error) {
```

Comment:
I'd suggest using `returnErr` name, so that it's not confused with other `err` usages in the method.

Reply:
I have renamed the output error, as well as some shadowed usages of `err` in the code, in order to remove confusion.